src/test/scala/com/fulcrumgenomics/umi/ExtractUmisFromBamTest.scala

```diff
@@ -383,6 +384,23 @@ class ExtractUmisFromBamTest extends UnitSpec with OptionValues {
     }
   }
 
+  it should "extract a single UMI from read one and leave read two untouched" in {
+    val builder = new SamBuilder(readLength=50)
+    val Seq(r1In, r2In) = builder.addPair(name="q1", unmapped1=true, unmapped2=true, start1=0, start2=0)
+    val output = newBam
+    val umi = r1In.basesString.substring(0, 8)
+
+    new ExtractUmisFromBam(input=builder.toTempFile(), output=output, readStructure=Seq("8M+T", "+T"), molecularIndexTags=Seq("A1")).execute()
+    val recs = readBamRecs(output)
+    recs should have size 2
+    val r1Out = recs.find(_.firstOfPair).getOrElse(fail("Couldnt find R1 in the output"))
+    val r2Out = recs.find(_.secondOfPair).getOrElse(fail("Couldnt find R2 in the output"))
+    r1Out.length shouldBe 42
+    r2Out.length shouldBe 50
+    r1Out.get("A1") shouldBe Some(umi)
+    r2Out.get("A1") shouldBe Some(umi)
```

Comment:
I'd prefer mixing in scala test `OptionValues` into the test class and then using:
```
r1Out.get("A1").value shouldBe umi
r2Out.get("A1").value shouldBe umi
```

I find using `.value` when we expect the `Option` to be defined to present a clearer error message when the test fails.